pkg/apis/app/v1beta1/kogitoinfra_types.go

```diff
@@ -51,6 +51,13 @@ type KogitoInfraSpec struct {
 	// Resource for the service. Example: Infinispan/Kafka/Keycloak.
 	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
 	Resource Resource `json:"resource,omitempty"`
+
+	// +optional
+	// +mapType=atomic
+	// Optional properties which would be needed to setup correct runtime/service configuration, based on the resource type.
+	// For example, MongoDB will require `username` and `database` as properties for a correct setup, else it will fail
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	InfraProperties map[string]string `json:"config,omitempty"`
```

Comment:
Question: why we have `InfraProperties` in the attribute but `config` in the CRD? If you're going to rename the type, you could also rename the CRD interface as well.

Reply:
woops. good catch. gonna update this.